core/src/main/java/com/adobe/granite/translation/connector/bootstrap/core/impl/BootstrapTranslationServiceImpl.java

```diff
@@ -282,60 +319,121 @@ public void addTranslationJobComment(String strTranslationJobID, Comment comment
 
     @Override
     public InputStream getTranslatedObject(String strTranslationJobID, TranslationObject translationObj)
-        throws TranslationException {
-        log.trace("BootstrapTranslationServiceImpl.getTranslatedObject");
-        return bootstrapTmsService.getTmsObjectTranslatedInputStream(strTranslationJobID, getObjectPath(translationObj));
+      throws TranslationException {
+      log.trace("BootstrapTranslationServiceImpl.getTranslatedObject");
+      String labels = String.format("%s,status=TRANSLATED", strTranslationJobID);
+      String objectPath = String.format("%s.%s", getObjectPath(translationObj), exportFormat);
+      try {
+        SourceFile[] files = liltApiClient.getFiles(labels);
+        // loop backwards through the list since the most recent files will be at the end.
+        for (int i = files.length - 1; i >= 0; i--) {
+          SourceFile file = files[i];
+          if (Objects.equals(file.name, objectPath)) {
+            log.warn("Downloading the translated lilt file {}", file.id);
+            String translation = liltApiClient.downloadFile(file.id);
+            return new ByteArrayInputStream(translation.getBytes());
+          }
+        }
+      } catch (Exception e) {
+        log.warn("error during getTranslatedObject {}", e);
+      }
+      log.info("No translations are available to download for {}", objectPath);
+      return null;
+    }
+
+    public Optional<String> getFileExtension(String filename) {
+      return Optional.ofNullable(filename)
+        .filter(f -> f.contains("."))
+        .map(f -> f.substring(filename.lastIndexOf(".") + 1));
     }
 
     @Override
     public String uploadTranslationObject(String strTranslationJobID, TranslationObject translationObject)
-        throws TranslationException {
-
-        InputStream inputStream;
-        log.trace("Using {} InputStream", exportFormat);
-        if (exportFormat.equalsIgnoreCase(BootstrapConstants.EXPORT_FORMAT_XLIFF_1_2)) {
-        	inputStream = translationObject.getTranslationObjectXLIFFInputStream("1.2");
-        } else if (exportFormat.equalsIgnoreCase(BootstrapConstants.EXPORT_FORMAT_XLIFF_2_0)) {
-        	inputStream = translationObject.getTranslationObjectXLIFFInputStream("2.0");
-        } else {
-        	inputStream = translationObject.getTranslationObjectXMLInputStream();
+      throws TranslationException {
+
+      InputStream inputStream;
```

Comment:
I'm not sure there's actually bug here, but there's the potential for the class of bug where the input stream is never closed and you leak either an open socket or an open file handle if an exception is thrown from the wrong place in this method (or another method that it calls), before the stream contents are consumed by `uploadFile()` (which probably closes the stream).

The old-fashioned way to deal with this is to wrap everything below this line in a try/finally block:
```java
InputStream inputStream;
try {
  // do stuff
}
finally {
  if (inputStream != null) {
   inputStream.close();
}
```

These days you should just use the "try with resources" feature, which makes the finally implicit.  However, do to this you will want to split the various things that can produce the inputStream out into a separate method.  You can also fold in your existing try/catch block here:

```java

// Define getStreamFromTranslationObject elsewhere.  This syntactic structure defines an implicit finally
// that will automatically close the InputStream (or anything else declared this way that implements
// the AutoCloseable interface.
try (InputStream inputStream = getStreamFromTranslationObject(translationObject, exportFormat) {
  // ....
  String labels = String.format("%s,status=READY", strTranslationJobID);
  liltApiClient.uploadFile(objectPath, labels, inputStream);
}
catch (Exception e) {
  log.warn("error during uploadTranslationObject", e);
}
```